scripts/verify-test-resource.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+set -o nounset # don't allow use of undefined variables
+set -o pipefail # if any single command in a pipe chain exits non-zero, set the exit status to that
+
+# Set environment variable for GitHub action to recognise colours
+export TERM=xterm
+
+# Colours for stdout
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+WHITE=$(tput setaf 7)
+NC=$(tput sgr0)
+RED_BG=$(tput setab 1)
+
+# Path to test resources
+TEST_RESOURCES_PATH=src/test/resources
+
+# Temporary directory for saving compiled classes
+TEMP_DIR_COMPILED_RESOURCES=$(mktemp -d)
+echo "Created $TEMP_DIR_COMPILED_RESOURCES for saving compiled files temporarily."
+echo -e "It will be removed after the script has executed with any exit code.\n"
+
+# Delete the temporary directory as it served its purpose
+cleanup() {
+  echo -e "\nRemoving $TEMP_DIR_COMPILED_RESOURCES directory ..."
+  rm -rf "$TEMP_DIR_COMPILED_RESOURCES"
+  echo "Done."
+}
+
+# Make sure the temporary directory is deleted even if the script aborts
+trap cleanup EXIT
+
+# Global variables which will be modified as the script runs
+status=0
+file_count=0
+compilation_errors=0
+
+# Verify if the resources compiles
+compile() {
+  local file=$1 # path to program in /tmp or equal to actual path
+  local actual_path=$2 # actual path of the test resource
+  if javac "$file" -d "$TEMP_DIR_COMPILED_RESOURCES"; then
+    echo -e "${actual_path}:${GREEN} Compiled successfully!${NC}"
+  else
+    echo -e "${RED}There were compilation errors in ${WHITE}${RED_BG}${actual_path}${NC}"
+    status=1
+    compilation_errors=$((compilation_errors+1))
+  fi
+}
+
+# Rename file according to the first public class inside
+get_and_rename_file() {
+  local file=$1
+  REGEX_PATTERN="(?<=public\sclass\s)[A-Z$_][\w$]*"
```

Comment:
This is the regex pattern I have written to search for the name of the public class.

Reply:
Actually, I'm surprised the regex works for the package private classes in the CI run, because it shouldn't. And it doesn't on my machine.

Reply:
I think the Java file name should match the **public class** name only. If there is not public class/interface inside the file,  the name can be any. The regex here does nothing if it doesn't find a public class. I made it that way because `javac` was compiling a file with one package private class even though the name did not match.

Reply:
Ah, yes, you're right, the file naming convention only applies to a public type. Funny, I didn't know that!

That leaves 5 ways in which the regex breaks :)

With that said, the very easiest solution would be to just... not have public types in the test files? :)

Reply:
> That leaves 5 ways in which the regex breaks :)

4, actually. `\w` contains the digits from `0-9` as well but the i18n is not taken care of so yeah, it might break there. Anyway, modifying this RegEx did not seem feasible as the lookbehinds need to be of fixed width but we have variable length keywords like - `class`, `interface`, etc. So I just decided to go with solution no. 3.

> just... not have public types in the test files? :)

I thought of that too but that is more of a workaround than a solution. Suppose we would want to test for an Update Operation which updates the access modifier from `private` to `public`. Our test resources won't compile in that case. The 3rd solution which you proposed works quite well and seems most elegant to me :)